### PR TITLE
ec2: Fix documentation

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
lib/ansible/modules/cloud/amazon/ec2.py

### CODE REVIEW

**Comment by Akasurde** [comment]
@goneri @wenottingham Could you please review this ? Thanks.

**Comment by goneri** [comment]
LGTM

File: lib/ansible/modules/cloud/amazon/ec2.py [hunk 1]
```diff
@@ -29,124 +29,149 @@
...
-      - security group id (or list of ids) to use with the instance.
+      - Security group id (or list of ids) to use with the instance.
+    type: list
   region:
     version_added: "1.2"
     description:
       - The AWS region to use.  Must be specified if ec2_url is not used.
-        If not specified then the value of the AWS_REGION or EC2_REGION environment variable, if any, is used.
-        See U(https://docs.aws.amazon.com/general/latest/gr/rande.html#ec2_region).
+      - If not specified then the value of the AWS_REGION or EC2_REGION environment variable, if any, is used.
```
Comment (gundalow) [inline lib/ansible/modules/cloud/amazon/ec2.py]:
```suggestion
      - If not specified then the value of the C(AWS_REGION) or C(EC2_REGION) environment variable, if any, is used.
```

File: lib/ansible/modules/cloud/amazon/ec2.py [hunk 2]
```diff
@@ -29,124 +29,149 @@
...
-      - instance type to use for the instance, see U(https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-types.html).
+      - Instance type to use for the instance, see U(https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-types.html).
     required: true
+    type: str
   tenancy:
     version_added: "1.9"
     description:
       - An instance with a tenancy of "dedicated" runs on single-tenant hardware and can only be launched into a VPC.
-        Note that to use dedicated tenancy you MUST specify a vpc_subnet_id as well. Dedicated tenancy is not available for EC2 "micro" instances.
+      - Note that to use dedicated tenancy you MUST specify a vpc_subnet_id as well.
```
Comment (gundalow) [inline lib/ansible/modules/cloud/amazon/ec2.py]:
```suggestion
      - Note that to use dedicated tenancy you MUST specify a I(vpc_subnet_id) as well.
```

File: lib/ansible/modules/cloud/amazon/ec2.py [hunk 3]
```diff
@@ -29,124 +29,149 @@
...
+      - Dedicated tenancy is not available for EC2 "micro" instances.
     default: default
     choices: [ "default", "dedicated" ]
+    type: str
   spot_price:
     version_added: "1.5"
     description:
-      - Maximum spot price to bid, If not set a regular on-demand instance is requested. A spot request is made with this maximum bid.
-        When it is filled, the instance is started.
+      - Maximum spot price to bid, If not set a regular on-demand instance is requested.
```
Comment (jillr) [inline lib/ansible/modules/cloud/amazon/ec2.py]:
```suggestion
      - Maximum spot price to bid. If not set, a regular on-demand instance is requested.
```

File: lib/ansible/modules/cloud/amazon/ec2.py [hunk 4]
```diff
@@ -29,124 +29,149 @@
...
     description:
-      - enable detailed monitoring (CloudWatch) for instance.
+      - Enable detailed monitoring (CloudWatch) for instance.
     type: bool
     default: 'no'
   user_data:
     version_added: "0.9"
     description:
-      - opaque blob of data which is made available to the ec2 instance.
+      - Opaque blob of data which is made available to the ec2 instance.
```
Comment (jillr) [inline lib/ansible/modules/cloud/amazon/ec2.py]:
```suggestion
      - Opaque blob of data which is made available to the EC2 instance.
```

**Review by jillr — DISMISSED** [review]

**Review by gundalow — APPROVED** [review]

**Pull request merged**